sql/core/src/test/scala/org/apache/spark/sql/util/ExecutionListenerManagerSuite.scala

```diff
@@ -69,3 +86,23 @@ private object CountingQueryExecutionListener {
   val INSTANCE_COUNT = new AtomicInteger()
 
 }
+
+private class SQLConfQueryExecutionListener(sqlConf: SQLConf) extends QueryExecutionListener {
```

Comment:
can't we just do `val conf = SparkSession.active.sessionState.conf` in the class body?

Reply:
> can't we just do `val conf = SparkSession.active.sessionState.conf` in the class body?

Since when we construct QueryExecutionListener, it's during the building process of `SessionState`, if we call `sessionState.conf`, it will throw `StackOverFlow`

Reply:
Can we delay the creation of `QueryExecutionListener` to after the session is built?

Reply:
> Can we delay the creation of `QueryExecutionListener` to after the session is built?

If so, we need to remove this parameter from SessionState, and make it as lazy. it's worth to do this?